### PR TITLE
Fix use of imageName and uri.GetName when transport is empty

### LINKED ISSUE #2681: singularity pull error with cache check directory
## Version of Singularity:

HEAD of master, 3.1.0-rc1.2.g3b402625

### Expected behavior

Pull image from library to local file.

### Actual behavior

```
 ./builddir/singularity -d -t ~/.singularity/sylabs-token-local pull alpine:latest
DEBUG   [U=501,P=83426]    apiGet()                      apiGet calling https://library.sylabs.io/v1/images///alpine:latest
DEBUG   [U=501,P=83426]    updateCacheSubdir()           Caching directory set to /Users/mikegray/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8
DEBUG   [U=501,P=83426]    updateCacheSubdir()           Caching directory set to /Users/mikegray/.singularity/cache/library
DEBUG   [U=501,P=83426]    updateCacheSubdir()           Caching directory set to /Users/mikegray/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8
DEBUG   [U=501,P=83426]    updateCacheSubdir()           Caching directory set to /Users/mikegray/.singularity/cache/library
FATAL   [U=501,P=83426]    pullRun()                     unable to check if /Users/mikegray/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8 exists: read /Users/mikegray/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8: is a directory
```

### Steps to reproduce behavior

Pull source code locally. Build on macOS Mojave - `mconfig`, `make -C builddir/`, run `singularity pull` command as above. I also tried deleting the cache directory  `rm -rf ~/.singularity/cache`, but the problem still occurs.

### PR DESCRIPTION
**Pull request by kzap**
Fixes #2681

Problem discussed in https://github.com/sylabs/singularity/issues/2681#issuecomment-462235773

> Still would like to add some tests covering the issue encountered

## Manual Testing
```sh
$ ./singularity -d pull alpine:latest
DEBUG   [U=502,P=69854]    apiGet()                      apiGet calling https://library.sylabs.io/v1/images///alpine:latest
DEBUG   [U=502,P=69854]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8
DEBUG   [U=502,P=69854]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library
DEBUG   [U=502,P=69854]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8
DEBUG   [U=502,P=69854]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library
```

identical to:
```sh
$ ./singularity -d pull library://alpine:latest
DEBUG   [U=502,P=69986]    apiGet()                      apiGet calling https://library.sylabs.io/v1/images///alpine:latest
DEBUG   [U=502,P=69986]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8
DEBUG   [U=502,P=69986]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library
DEBUG   [U=502,P=69986]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library/sha256.69ce2a3dcc6d3e559e20ced0df251046ee6ecff390a945d856fe0dcb3bcb3ce8
DEBUG   [U=502,P=69986]    updateCacheSubdir()           Caching directory set to /Users/andremarcelo-tanner/.singularity/cache/library
```

### CODE REVIEW

**Comment by kzap** [comment]
ok tests are failing with
```sh
--- FAIL: TestPull (0.77s)
    --- FAIL: TestPull/Pull_Library (0.77s)
        --- FAIL: TestPull/Pull_Library/RunScript (0.03s)
            build_test.go:50: FATAL:   failed to retrieved path for /home/testuser/rpmbuild/BUILD/singularity-3.1.0/gopath/src/github.com/sylabs/singularity/cmd/singularity/test_pull.sif: lstat /home/testuser/rpmbuild/BUILD/singularity-3.1.0/gopath/src/github.com/sylabs/singularity/cmd/singularity/test_pull.sif: no such file or directory
```

I broke the ability to specify the local name when saving the file

File: cmd/internal/cli/pull.go [hunk 1]
```diff
@@ -91,7 +91,11 @@ func pullRun(cmd *cobra.Command, args []string) {
 	if PullImageName == "" {
 		name = args[0]
```
Comment (kzap) [inline cmd/internal/cli/pull.go:92]: would really like to rename this to `dstFileName` as the name saved in the cache dir can be different from the file name saved by the user

File: cmd/singularity/pull_test.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ func TestPull(t *testing.T) {
 		{"Pull_Docker", "docker://alpine:3.7", true, "", imagePath, true},   // https://hub.docker.com/
 		{"Pull_Shub", "shub://GodloveD/busybox", true, "", imagePath, true}, // https://singularity-hub.org/
 		{"PullWithHash", "library://alpine:sha256.af5e6b93dcfd08b7b46d6158c31ae5b6c2b4241c169b65595659ad9feac6f761", true, "", imagePath, true},
+		{"PullWithoutTransportProtocol", "alpine:3.7", true, "", imagePath, true},
```
Comment (kzap) [inline cmd/singularity/pull_test.go:51]: new test to pull without a `library://` protocol

**Review by ArangoGutierrez — APPROVED** [review]
LGTM
Does fix the target issue

**Comment by ArangoGutierrez** [comment]
@mikegray  Could you check if this patch solves issue #2681 , so we can include it as part of 3.1-rc2 fixes. thanks!

**Comment by mikegray** [comment]
lgtm

**Comment by ArangoGutierrez** [comment]
@GodloveD pls merge as part of `3.1-RC fixes`

**Pull request merged**